### PR TITLE
Event Service API to .Net Core 3.0

### PR DESCRIPTION
**Pull request by Rossall82**
- Move ReCreateLocalDb to Database Helpers
- Implement PostEvent controller method so requested event is published
- Unit test to cover publishing an event

### CODE REVIEW

File: src/Common/Common.Messages/Events/GregTestEvent.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+using NServiceBus;
+
+namespace Common.Messages.Events
+{
+    public class GregTestEvent : IEvent, ICorrelate
+    {
+        public Guid CorrelationId { get; set; }
+        public string Gregio { get; set; } = "mooo";
+    }
+}
```
Comment (SamirHassoun) [inline src/Common/Common.Messages/Events/GregTestEvent.cs]: GregTestEvent? :-)

File: src/EventServices/EventService.UnitTests/EventControllerTests.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        {
+            var testableSession = new TestableMessageSession();
+
+            var eventController = new Controllers.EventController(testableSession);
+
+            var theEvent = new GregTestEvent
+            {
+                CorrelationId = Guid.NewGuid(),
+                Gregio = "Wow"
+            };
```
Comment (SamirHassoun) [inline src/EventServices/EventService.UnitTests/EventControllerTests.cs]: GregTestEvent? :-)

File: src/EventServices/EventService/EventService.csproj [hunk 3]
```diff
@@ -1,31 +1,30 @@
...
-      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
-    </Content>
+    <PackageReference Include="Microsoft.AspNetCore.Diagnostics.HealthChecks" Version="2.2.0" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="3.0.0" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.Tools" Version="3.0.0">
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+    </PackageReference>
+    <PackageReference Include="Microsoft.Extensions.Logging.Debug" Version="3.0.0" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.3.0" />
```
Comment (UKHOGreg) [inline src/EventServices/EventService/EventService.csproj]: Do we definitely need this package in EventService?

File: src/Common/Common.Messages/Events/GregTestEvent.cs (hunk 1 above)
Comment (UKHOGreg) [reply src/Common/Common.Messages/Events/GregTestEvent.cs]: mooo!

File: src/EventServices/EventService/Controllers/DefaultController.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+            var publishOptions = new PublishOptions();
+            await _messageSession.Publish(message, publishOptions).ConfigureAwait(false);
+            //var sendOptions = new SendOptions();
+            //sendOptions.SetDestination("UKHO.TaskManager.SourceDocumentCoordinator");
+            //await _messageSession.Send(message, sendOptions)
+            //    .ConfigureAwait(false);
+            return "Message sent to endpoint";
+        }
+    }
+}
```
Comment (UKHOGreg) [inline src/EventServices/EventService/Controllers/DefaultController.cs]: Should we remove this controller, used for manual testing only?

File: src/EventServices/EventService/Startup.cs [hunk 5]
```diff
@@ -1,121 +1,166 @@
...
+
+            app.UseAuthorization();
+
+            app.UseSwagger();
+            app.UseSwaggerUI(c =>
             {
-                //TODO: Enable production exception handling (https://docs.microsoft.com/en-us/aspnet/core/fundamentals/error-handling)
-                // app.UseExceptionHandler("/Home/Error");
-            }
+                c.SwaggerEndpoint("/swagger-original.json", "SDRA API Original");
```
Comment (UKHOGreg) [inline src/EventServices/EventService/Startup.cs]: `"SDRA API Original"` doesn't seem right!

File: src/EventServices/EventService/Controllers/DefaultController.cs (hunk 4 above)
Comment (Rossall82) [reply src/EventServices/EventService/Controllers/DefaultController.cs]: Yep, done :)

**Pull request merged**